### PR TITLE
Add cluster-checker tool

### PR DESCRIPTION
**Pull request by tardieu**
This tool diagnoses the state of an MLBatch cluster looking for common issues and producing a gpu allocation summary per namespace and for the entire cluster.

### CODE REVIEW

File: tools/cluster-checker/checker.js [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  // compute sum of regular containers
+  for (const container of pod.spec.containers) {
+    if (container.resources?.requests?.['nvidia.com/gpu'] > 0) {
+      gpus += parseInt(container.resources?.requests?.['nvidia.com/gpu'])
+    } else if (container.resources?.limits?.['nvidia.com/gpu'] > 0) {
+      gpus += parseInt(container.resources?.limits?.['nvidia.com/gpu'])
+    }
+  }
+  // compute max with init containers
+  for (const container of pod.spec.initContainers ?? []) {
```
Comment (dgrove-oss) [inline tools/cluster-checker/checker.js:81]: init containers run in parallel.  Don't you need to sum their GPU usage together and then do the max operation outside the loop?

**Review by dgrove-oss — APPROVED** [review]
One small question, but otherwise looks great

File: tools/cluster-checker/checker.js (hunk 1 above)
Comment (tardieu) [reply tools/cluster-checker/checker.js:81]: This is not my understanding of init containers. See here: https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#understanding-init-containers.

Comment (dgrove-oss) [reply tools/cluster-checker/checker.js:81]: So much for trusting my memory... I really thought we had multiple init containers somewhere in the past in KAR that ran in parallel... comment withdrawn.

**Pull request merged**